cmd/juju/commands/deploy.go

```diff
@@ -190,13 +200,31 @@ func (c *DeployCommand) Run(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 	defer csClient.jar.Save()
-
 	repoPath := ctx.AbsPath(c.RepoPath)
-	curl, repo, err := resolveCharmStoreEntityURL(c.CharmName, csClient.params, repoPath, conf)
+
+	// Handle local bundle paths.
+	f, err := os.Open(c.CharmOrBundle)
+	if err == nil {
+		defer f.Close()
+		bundleData, err := charm.ReadBundleData(f)
```

Comment:
Perhaps check whether the file is a directory here and produce an appropriate error here?